cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -96,7 +98,20 @@ private void setMetadata(MetadataEntry metadataEntry) {
    * @param value The metadata value to be added
    */
   public void setProperty(Id.NamespacedId targetId, String key, String value) {
-    setMetadata(new MetadataEntry(targetId, key, value));
+    setProperty(targetId, key, value, IndexerFactory.getIndexer(IndexerType.DEFAULT_VALUE_INDEXER));
+  }
+
+  /**
+   * Sets a metadata property for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId The target Id: app-id(ns+app) / program-id(ns+app+pgtype+pgm) /
```

Comment:
I don't think you need the info in parentheses.
Simply specifying `ProgramId` means ns + app + pgtype + pgmName, because of how `ProgramId` class is defined, right?